src/index.js

```diff
@@ -239,18 +278,22 @@ const reduceRule = (rule, attributes) => {
   if (Object.keys(result).length === 0) {
     return true;
   }
+
   return result;
 };
 
 const reduceRules = (rules, attributes) => {
+  const attributesClone = cloneDeep(attributes);
+
   const result = [];
 
   if (rules === true) {
     return true;
   }
 
-  for (const rule of rules) {
-    const reducedRule = reduceRule(rule, attributes);
+  for (const rule of cloneDeep(rules)) {
```

Comment:
This function was cloning `rule` in certain cases, which was causing the actual `policy` to be mutated.